### DISCUSSION TITLE
Windows deployment of Nebula, how to do?

### DISCUSSION THREAD

**Discussion by Skreabengt** [slackhq/nebula#1089] [Q&A] [answered]
I'm looking at Nebula for the first time and mainly to setup a site-to-site communication for IPv6 between two sites with native IPv6 and dual stack, which unfortunatley cannot be done between the routers I have anymore. I will do it in Windows and I suppose the nebula.exe and nebula-cert.exe files need to be to be deployed in some way to respond to commands in powershell or through ssh.

Most examples with deployments e.g. in Youtube seems to be for Linux, and I am afraid that this is a really stupid question, but I have to ask in anyway.

I found the following instruction on one GitHub repository: "On Windows, execute (as Administrator) the deploy.bat batch file which will ask for a target directory where all the required files will be placed, and will install and start a Windows service"

I have searched but cannot find any bacth file, so is there any batch file or is deployment done differently?

Background; I have had VPN IPsec tunnels in IPv4 since long and was about to configure VPN tunnels for IPv6. However, it doesn't matter how hard I try it turn out, since VPN IPsec for IPv6 is no longer supported on Edgerouters v.2.09  hotfix 7 according to Ubiquiti customer support.

**Comment by nbrownus** [comment, marked as answer]
The easiest method would be to get the nebula binary from most recent [release](https://github.com/slackhq/nebula/releases) and using an administrator command prompt or powershell running `nebula.exe service install` and `nebula.exe service start`. This will establish a windows service and start it. The config location will default to `config.yaml` in the directory you are running nebula from, you can override it with a `-config` flag.

[Here](https://nebula.defined.net/docs/guides/quick-start/) are some guides and general documentation to help get you started.

The main thing to note before you begin is that nebula is really intended to be a point to point mesh vpn. It does support site to site via what we call unsafe routes though this does leave some of the routing/masquerading setup to the administrator. Nebula also does not currently support ipv6 on the overlay.

**Reply by hmoffatt** [reply]
That should be `nebula.exe -service install` I think? At least, that's what worked for me with the current release (1.8.2).

**Comment by Skreabengt** [comment]
I will look at other ways forward, since nebula don't support IPv6 and is not intended for site-to-site.

It sounds quite simple to install and start nebula from powershell, and will still be useful information, thanks! I will instead use it for connecting a laptop on travel and for the children, which are students living away from home.

The site-to-site VPN can likely be solved through IPsec in OPNsense and then with tunnels in both IPv6 and IPv4.
